File: LeagueItemExplorerJAR/LeagueItemExplorerJAR/src/main/java/com/app/LeagueItemExplorerJAR/models/ChampionDataExtended.java
package com.app.LeagueItemExplorerJAR.models;

import lombok.Data;

@Data
public class ChampionDataExtended {
    private String id;
    private String key;
    private String name;
    private String title;
    private ChampionImage image;
    private String[] tags;
    private String partype;
    private String lore;
    private String blurb;
    private String[] allytips;
    private String[] enemytips;
    private ChampionStats stats;
    private ChampionPassive passive;
    private ChampionSpell[] spells;
}

File: LeagueItemExplorerFront/src/app/models/champion-data-extended.ts
import { ChampionImage } from "./champion-image";
import { ChampionInfo } from "./champion-info";
import { ChampionPassive } from "./champion-passive";
import { ChampionSpell } from "./champion-spell";
import { ChampionStats } from "./champion-stats";

export interface ChampionDataExtended {
    id: string;
    key: string;
    name: string;
    title: string;
    image: ChampionImage;
    tags: string[];
    partype: string;
    info: ChampionInfo;
    lore: string;
    blurb: string;
    allytips: string[];
    enemytips: string[];
    stats: ChampionStats;
    passive: ChampionPassive;
    spells: ChampionSpell[];
}